ast/src/analyzed/mod.rs

```diff
@@ -107,6 +107,34 @@ impl<T> Analyzed<T> {
             + 1
     }
 
+    pub fn next_id_for_kind(&self, poly_type: PolynomialType) -> u64 {
+        match poly_type {
+            PolynomialType::Committed => {
+                self.committed_polys_in_source_order()
+                    .map(|(symbol, _)| symbol.id)
```

Comment:
should also use array_elements, shouldn't it?

Reply:
True, fixed.